src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java

```diff
@@ -279,49 +279,70 @@ public static void scan(Run build, PrintStream scanLog) {
      * @param buildNum - Build object
      * @param buildUrl - Full URL of build
      * @param scanLog - PrintStream for the build log
-     * @return boolean true if message successfully created, false otherwise
+     * @return String Slack message with failure name, category and description if message successfully created, null otherwise
      */
-    public static boolean createSlackMessage(List<FoundFailureCause> foundCauseList,
+    public static String createSlackMessage(List<FoundFailureCause> foundCauseList,
             boolean notifySlackOfAllFailures, List<String> slackFailureCauseCategories,
             String buildName, String buildNum, String buildUrl, PrintStream scanLog) {
         boolean notifySlackOfFailure = false;
         StringBuilder bufBuildFailCause = new StringBuilder();
 
         /* Check if one of the failure causes for the build matches those specified in plugin's slack settings. */
         for (FoundFailureCause foundCause : foundCauseList) {
-            if (!notifySlackOfAllFailures) {
+            if (notifySlackOfAllFailures) {
+                //Add two new lines between found categories
```

Comment:
From what I can see this adds two new lines between each cause and not each category. Did you intend each category or each cause?

Reply:
I intended between failure causes will update the docs